lib/reek/context/attribute_context.rb

```diff
@@ -15,6 +15,15 @@ def full_comment
         send_expression.full_comment || ''
       end
 
+      def instance_method?
+        true
+      end
```

Comment:
How about:

```Ruby
class AttributeContext
  @method_properties = {
    instance_method: true,
    singleton_method: false
  }
  @method_properties.keys.each do |predicate|
    define_singleton_method "#{predicate}?" do
      @method_properties.fetch predicate
    end
  end
end
```

?

That is super-hard to read, completely overengineered but looks like we know what the hell we are doing here!

Ok, on second thought let's go with your solution. #no-yolo